frontend/src/components/Navbar.js

```diff
@@ -4,66 +4,89 @@ import { Navbar, Nav, Button } from 'react-bootstrap'
 import Modal from './SignInUpModal';
 import style from 'bootstrap/dist/css/bootstrap.css';
 import Firebase from 'firebase';
+import { changeMapState } from "../actions/index";
+import { connect } from "react-redux";
+import store from "../store/index";
+import { Provider } from "react-redux";
 
-export default class TopNavbar extends React.Component {
+function mapDispatchToProps(dispatch) {
+  return {
+    changeMapState: mapState => dispatch(changeMapState(mapState))
+  };
+}
+
+const mapStateToProps = state => {
+  return { articles: state.articles };
+}
+
+class TopNavbar extends React.Component {
   constructor(props) {
     super(props);
 
-    this.state= {
-      loggedIn: props.loggedIn,
-      history: props.history,
+    if (this.props.articles[0]) {
+      this.state = {
+        loggedIn: this.props.articles[0].loggedIn,
+        plusCode: this.props.articles[0].plusCode,
+        credentials: this.props.articles[0].credentials
+      }
+    } else {
+      this.state = {
+        loggedIn: false,
+      }
     }
+  }
 
-    var credentials;
-
-    if (props.credentials) {
-      this.credentials = props.credentials;
-    } else if (props.history.location.state){
-      if (props.history.location.state.credentials) {
-        this.credentials = props.history.location.state.credentials;
-      }
+  updateReduxWhenLoggingOut() {
```

Comment:
Since this function is for logging out, how about `logOutAndUpdateRedux` instead? "WhenLoggingOut" sounds like it's setting up some sort of listener